### ISSUE TITLE
Docker: mount path must be absolute.

### ISSUE THREAD

**Issue by Pitometsu** [commercialhaskell/stack#2324]
How can I specify mount point for docker in `stack.yaml` for file in current user `$HOME` without hardcoding full path? Or generate mount points by script like bash?

**Comment by Pitometsu** [comment]
Would be nice some preprocess language, e.g. ability to script on bash in `stack.yaml`.

**Comment by sjakobi** [comment]
> How can I specify mount point for docker in stack.yaml for file in current user $HOME without hardcoding full path?

That idea is tracked in https://github.com/commercialhaskell/stack/issues/1375.



> Would be nice to have some preprocess language, e.g. ability to script on bash in stack.yaml.

I think that idea is not so compatible with the goal of keeping the `stack.yaml` format simple. It also sounds like it would make the configuration less reproducible

But I think that `stack config set` could be used to this.

**Comment by Pitometsu** [comment]
@sjakobi, thank you for response
> But I think that `stack config set` could be used to this.

```
 $ stack config set extra-deps "[ipprint-0.6, sr-extra-1.46.3.2]"
stack: parse stack config set field: only set resolver is implemented
```

**Comment by sjakobi** [comment]
> $ stack config set extra-deps "[ipprint-0.6, sr-extra-1.46.3.2]"
> stack: parse stack config set field: only set resolver is implemented

Yeah, `stack config set` currently has only very rudimentary functionality.

#115 contains a general discussion of the `stack config` command, so I guess that's the right place for discussing the syntax for the extension that you would like.

I'm afraid I'm not really familiar with docker and stack's docker integration, so maybe you can give a more concrete example of the functionality that you need?

**Comment by Pitometsu** [comment]
Just ability to create symlinks more flexable, depend on host environment without path pardcoding. And `stack config` nice feature, would be glad to use described functionality when it will be implemented just as described in issue above. Thanks.
